pipelined/bevy_sprite2/src/render/mod.rs

```diff
@@ -476,26 +477,26 @@ pub fn queue_sprites(
         );
         for mut transparent_phase in views.iter_mut() {
             for (entity, batch) in sprite_batches.iter_mut() {
-                image_bind_groups
-                    .values
-                    .entry(batch.handle.clone_weak())
-                    .or_insert_with(|| {
-                        let gpu_image = gpu_images.get(&batch.handle).unwrap();
-                        render_device.create_bind_group(&BindGroupDescriptor {
-                            entries: &[
-                                BindGroupEntry {
-                                    binding: 0,
-                                    resource: BindingResource::TextureView(&gpu_image.texture_view),
-                                },
-                                BindGroupEntry {
-                                    binding: 1,
-                                    resource: BindingResource::Sampler(&gpu_image.sampler),
-                                },
-                            ],
-                            label: Some("sprite_material_bind_group"),
-                            layout: &sprite_pipeline.material_layout,
-                        })
-                    });
+                let gpu_image = gpu_images.get(&batch.handle).unwrap().clone();
+                match image_bind_groups.values.entry(batch.handle.clone_weak()) {
+                    Entry::Occupied(mut already_existing) => {
+                        let (group, image) = already_existing.get_mut();
+                        // Can't compare GPUImages directly, this has to be good enough
+                        if image.texture_view.id() != gpu_image.texture_view.id()
```

Comment:
This adds additional "per entity" work to a pretty hot path. I think it is better to just extract Image AssetEvents into a resource (EventReader params work as expected in the Extract step), then remove "changed" or "removed" handles from ImageBindGroups in the Prepare step. That has the added benefit of cleaning up bind groups we don't need anymore.